packages/image_picker/tizen/src/image_resize.cc

```diff
@@ -79,164 +87,123 @@ bool ImageResize::TransformImage(transformation_h transform_h,
+    std::transform(extension.begin(), extension.end(), extension.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
+    if (extension == "png") {
+      type = IMAGE_UTIL_PNG;
+    } else if (extension == "gif") {
+      type = IMAGE_UTIL_GIF;
+    } else if (extension == "bmp") {
+      type = IMAGE_UTIL_BMP;
+    }
+  }
+
+  image_util_encode_h handle = nullptr;
+  int ret = image_util_encode_create(type, &handle);
+  if (ret != IMAGE_UTIL_ERROR_NONE) {
+    LOG_ERROR("Failed to initialize image encode util: %s",
+              get_error_message(ret));
+    return false;
+  }
+
+  if (type == IMAGE_UTIL_JPEG) {
+    if (IsValidQuality()) {
+      LOG_DEBUG("Target image quality: %d", quality_);
+      int ret = image_util_encode_set_quality(handle, quality_);
       if (ret != IMAGE_UTIL_ERROR_NONE) {
-        LOG_ERROR("image_util_encode_set_quality fail! [%s]",
-                  get_error_message(ret));
+        LOG_ERROR("Failed to set image quality: %s", get_error_message(ret));
+        image_util_encode_destroy(handle);
         return false;
       }
     }
   } else {
-    LOG_DEBUG(
-        "image_picker: image quality option supports only JPG type.. "
-        "Returning the image with original quality");
+    LOG_INFO(
+        "image_picker: The image quality option only supports JPG type. "
+        "Returning the image with original quality.");
   }
 
-  LOG_DEBUG("dst_path : %s", dst_file.c_str());
-  int ret =
-      image_util_encode_run_to_file(encode_h, dst_image, dst_file.c_str());
+  LOG_DEBUG("Target image path: %s", path.c_str());
+  ret = image_util_encode_run_to_file(handle, image, path.c_str());
   if (ret != IMAGE_UTIL_ERROR_NONE) {
-    LOG_ERROR("image_util_encode_run_to_file fail! [%s]",
-              get_error_message(ret));
+    LOG_ERROR("Failed to encode image: %s", get_error_message(ret));
+    image_util_encode_destroy(handle);
     return false;
   }
 
+  image_util_encode_destroy(handle);
   return true;
 }
 
-bool ImageResize::Resize(const std::string& src_file, std::string& dst_file) {
-  LOG_DEBUG("source image path: %s", src_file.c_str());
-
-  if (max_width_ == 0 && max_height_ == 0 &&
-      (quality_ <= 0 || quality_ > 100)) {
-    return false;
-  }
-
-  // ===========================================================
-  image_util_image_h src_image = nullptr;
-  image_util_image_h dst_image = nullptr;
-  image_util_decode_h decode_h = nullptr;
-
-  int ret = image_util_decode_create(&decode_h);
-  if (ret != IMAGE_UTIL_ERROR_NONE) {
-    LOG_ERROR("image_util_decode_create fail! [%s]", get_error_message(ret));
-    return false;
-  }
-  bool is_decoded = DecodeImage(decode_h, src_image, src_file);
-  image_util_decode_destroy(decode_h);
-  if (!is_decoded) {
-    if (src_image) {
-      image_util_destroy_image(src_image);
-    }
+bool ImageResize::Resize(const std::string& source_path,
+                         std::string& dest_path) {
```

Comment:
prefer using pointer rather reference to represent output

Reply:
I already tried this but for some reason it didn't look very good to me. I'll try again.